src/main/java/graphql/nadel/ServiceExecutionHooks.java

```diff
@@ -0,0 +1,25 @@
+package graphql.nadel;
+
+import graphql.execution.ExecutionStepInfo;
+import graphql.execution.nextgen.result.RootExecutionResultNode;
+import graphql.language.Argument;
+import graphql.schema.GraphQLSchema;
+
+import java.util.List;
+
+public interface ServiceExecutionHooks {
+
+
+    default Object createServiceContext(Service service, ExecutionStepInfo topLevelStepInfo) {
+        return null;
+    }
+
+    default List<Argument> modifyArguments(Service service, Object serviceContext, ExecutionStepInfo topLevelStepInfo, List<Argument> arguments) {
+        return null;
+    }
+
+    default RootExecutionResultNode postServiceResult(Service service, Object serviceContext, GraphQLSchema overallSchema, RootExecutionResultNode resultNode) {
+        return resultNode;
+    }
+
+}
```

Comment:
From an API point of view - we will get more milage with XXXEnvironment single parameter style
Maybe
```
class ServiceExecutionHooksEnvironment {
   Service service, 
  Object serviceContext, 
  GraphQLSchema overallSchema,
  ExecutionStepInfo topLevelStepInfo
}
```

That is some might be null in the second calls

```
public interface ServiceExecutionHooks {


    default Object createServiceContext(ServiceExecutionHooksEnvironment env) {
        return null;
    }

    default List<Argument> modifyArguments(ServiceExecutionHooksEnvironment env, List<Argument> arguments) {
        return null;
    }

    default RootExecutionResultNode postServiceResult(ServiceExecutionHooksEnvironment env, RootExecutionResultNode resultNode) {
        return resultNode;
    }

}
```

Reply:
We are likely to add more parameters and hence break less code

Reply:
I agree, but it doesn't match really at the moment: serviceContext is not available on all three, neither is the topLevelStepInfo ....